onedr0p/flux-cluster-template
# feat: Change configure to mjs with google zx

This PR changes the current bash script of configure to a (bit) more structured and more maintainable script. At this moment it's a rewrite without to many changes (some QOL).

QOL changes:
- Checking if verify was run before configure
- Checking of config was changed between verify and configure
- Writing output to log for easier debugging/sharing
- Debug option
- Configuration checking (missing required fields, regex patterns)

To make it work:
1. pull branch
2. make sure you have latest node lts
3. run 'npm i'
4. run 'npm i -g zx'
5. run 'chmod +x configure.mjs'
6. run './configure.mjs' to see the usage

(woops forgot the PR)
